worlds/apquest/test/test_extra_starting_chest.py

```diff
@@ -0,0 +1,38 @@
+from test.param import classvar_matrix
+
+from .bases import APQuestTestBase
+
+# Sometimes, you might want to test something with a specific option disabled, and then with it enabled.
+# To do this efficiently, we can use param.classvar_matrix.
+# First, we prepare a list of every option(s combination) we want to test:
+extra_starting_chest_options = [
+    {"extra_starting_chest": False},
+    {"extra_starting_chest": True},
+]
+
+
+# Then, we decorate our TestCase with @classvar_matrix, passing our options into it.
+# Our TestCase will then be run once with the extra_starting_chest option disabled, and once with it enabled.
+# Remember: The WorldTestBase class has a ClassVar called "options", which is what we're setting here.
+# CAREFUL: Passing multiple ClassVars to classvar_matrix will run *all combinations*, not perform a "zip" operation.
+@classvar_matrix(options=extra_starting_chest_options)
+class TestExtraStartingChest(APQuestTestBase):
+    # Our extra starting chest only exists if the extra_starting_chest option is enabled, so let's verify that.
+    def test_extra_starting_chest_exists(self) -> None:
```

Comment:
Presumably, this whole matrix approach is only beneficial if there's a lot of shared test logic you want to be identical in all of the relevant option combinations, right? If so, shouldn't we put a placeholder comment like `# all the tests that don't depend on the options we're varying` to clarify that?

Reply:
Added a bit more text here